src/qibolab/instruments/rfsoc.py

```diff
@@ -232,13 +237,12 @@ def play(
 
                 serial = ro_pulse.serial
 
-                if average:
-                    results[ro_pulse.qubit] = results[serial] = AveragedResults.from_components(i_pulse, q_pulse)
+                if execution_parameters.acquisition_type is AcquisitionType.DISCRIMINATION:
+                    discriminated_shots = self.classify_shots(i_pulse, q_pulse, qubits[ro_pulse.qubit])
+                    result = execution_parameters.results_type(discriminated_shots)
                 else:
-                    shots = self.classify_shots(i_pulse, q_pulse, qubits[ro_pulse.qubit])
-                    results[ro_pulse.qubit] = results[serial] = ExecutionResults.from_components(
-                        i_pulse, q_pulse, shots
-                    )
+                    result = execution_parameters.results_type(i_pulse + 1j * q_pulse)
+                results[ro_pulse.qubit] = results[serial] = result
```

Comment:
In the first version of the driver I did it because I was told the qibocal uses sometimes the serial and sometime the qubit... Is this still the case @andrea-pasquale, @Edoardo-Pedicillo ?

Reply:
I remember that we already add this discussion with @AleCandido and @stavros11 a few months ago.
I don't remember the exact motivation, maybe it was related to the fact that drivers were not so stable a few months ago. Also, with the current implementation, especially in qibocal we tend to modify directly the pulses during the calibration routines therefore you may not recover your results correctly.
At the same time we cannot just have the qubit since we may want to do multiple measurements on the same qubit...